Touch Bar Simulator/util.swift

```diff
@@ -53,6 +84,76 @@ extension NSView {
 	}
 }
 
+final class AssociatedObject<T: Any> {
+	subscript(index: Any) -> T? {
+		get {
+			return objc_getAssociatedObject(index, Unmanaged.passUnretained(self).toOpaque()) as! T?
+		} set {
+			objc_setAssociatedObject(index, Unmanaged.passUnretained(self).toOpaque(), newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+		}
+	}
+}
+
+@objc
+protocol TargetActionSender: AnyObject {
+	var target: AnyObject? { get set }
+	var action: Selector? { get set }
+}
+
+extension NSControl: TargetActionSender {}
+extension NSMenuItem: TargetActionSender {}
+
+private final class ActionClosureCaller: NSObject {
+	@objc
+	func callClosure(_ sender: TargetActionSender) {
+		onAction?(sender)
+	}
+
+	var onAction: TargetActionSender.ActionClosure?
+}
+
+struct TargetActionSenderAssociatedKeys {
+	fileprivate static let caller = AssociatedObject<ActionClosureCaller>()
+}
+
+extension TargetActionSender {
+	typealias ActionClosure = ((TargetActionSender) -> Void)
+
```

Comment:
I prefer keeping the keys on the extension:

```swift
private struct AssociatedKeys {
		static let caller = AssociatedObject<ActionClosureCaller>()
```

Reply:
Swift won't let me nest a struct in a protocol extension.

Reply:
Damn, forgot about that limitation. Swift really needs namespaces...